docs/docs/crafty.md

```diff
@@ -3,4 +3,184 @@ id: crafty
+
+```
+import 'zos-lib/contracts/migrations/Initializable.sol';
+
+contract Crafty is Initializable {
+  // Standard Solidity constructor - will be removed from source code
+  function Crafty() public {
+    addRole(msg.sender, ROLE_ADMIN);
+  }
+
+  // Initializer for integration with ZeppelinOS
+  function initialize(address _initialAdmin) isInitializer public {
+    addRole(_initialAdmin, ROLE_ADMIN);
+  }
+  ...
+}
+```
+
+The `isInitializer` modifier will make sure your `initialize` method is only called once in the whole lifetime of your contract. But you must remember to explicitly call it, since unlike constructors, `initialize` methods are not automatically called at contract creation.
+
+With that change in place, we're all set to start using upgradeable instances of the `Crafty` contract! First, we `add` it to the project's contracts, and then `push` the compiled bytecode of the current version to the blockchain.
+
+```sh
+zos add Crafty
+zos push --network ropsten
+```
+
+We can now `create` new upgradeable instances, all of which will use the same deployed bytecode, but be entirely independent otherwise. The `initialize` function can be called in this same step, saving you the need to do it manually from a console.
+
+```sh
+zos create Crafty --init --args 0x0cbd7... --network ropsten
+> 0x31C4B...
+```
+
+The returned value is the address of the newly `create`d Crafty proxy, which is already initialized and can be safely used, with the peace of mind that it can later be upgraded at any point in time.
+
+### Making `CraftableToken` upgradeable
+
+This scenario is a bit more complex, since multiple ZeppelinOS contracts are being combined together, but it is nonetheless fairly easy to setup. Let's first recap what a `CraftableToken` is:
+
+1. It is detailed ERC20 token (i.e. has name and symbol)
+1. It supports the [EIP-1046](https://github.com/ethereum/EIPs/blob/master/EIPS/eip-1046.md) extension of the ERC20 standard (i.e. we'll add a `tokenURI` parameter and variable)
+1. It is mintable
+1. It has additional storage of the token's ingredients, which require some validation
+
+Luckily for us, OpenZeppelin provides contracts that will implement 1 and 3 for us: `DetailedERC20` and `MintableToken`. 2 and 4 we'll have to cover on our own, but the process is quite simple, and barely differs from what it would look like without ZeppelinOS.
+
+The EIP-1046 extension is very straightforward: we'll inherit from OpenZeppelin's `DetailedERC20` and add the required variable to our contract. The only difference is, once again, that we need to define an `initialize` method instead of a constructor, and call our base contracts' `initialize` methods in it.
```

Comment:
Perhaps clarify a bit more about why we're using `openzeppelin-zos` instead of `openzeppelin-solidity` here?